src/data/addressbook.go

```diff
@@ -0,0 +1,623 @@
+func (ab *DB) encryptContact(c *Contact) ([]byte, error) {
+	secType, err := ab.GetSecType()
+	if err != nil {
+		return nil, err
+	}
+	switch secType {
+	case Type1:
+		return c.MarshalBinary()
+	case Type2:
+		data, err := c.MarshalBinary()
+		if err != nil {
+			return nil, err
+		}
+		return toBase64(data), nil
+	case Type3:
+		return ab.encryptAESGCM(c)
+	}
+
+	return nil, fmt.Errorf("invalid security type")
+}
+
+// Decrypt a cipher message using a password with AES-GCM and return a Contact.
+func (ab *DB) decryptContact(cipherMsg []byte) (core.Contact, error) {
+	secType, err := ab.GetSecType()
+	if err != nil {
+		return nil, err
+	}
+	switch secType {
+	case Type1:
+		c := Contact{}
+		if err := c.UnmarshalBinary(cipherMsg); err != nil {
+			return nil, err
+		}
+
+		return &c, nil
+
+	case Type2:
+		c := Contact{}
+		data, err := fromBase64(cipherMsg)
+		if err != nil {
+			return nil, err
+		}
+		if err := c.UnmarshalBinary(data); err != nil {
+			return nil, err
+		}
+		return &c, nil
+	case Type3:
+		return ab.decryptAESGCM(cipherMsg)
+	}
+
+	return nil, errInvalidSecType
+}
+
+//
+func (ab *DB) verifyHash() error {
+	hash, err := ab.getHashFromConfig()
+	if err != nil {
+		return err
+	}
+	return bcrypt.CompareHashAndPassword(hash, ab.key)
+}
+
+// addressExists search an address in the list of contacts into the AddressBook.
+// If find the address return error, else return nil.
+func (ab *DB) addressExists(address core.StringAddress, contacts []core.Contact) error {
+	for _, v := range contacts {
+		c, ok := v.(*Contact)
+		if ok {
+			for _, addrs := range c.Address {
+				if bytes.Compare(addrs.GetValue(), address.GetValue()) == 0 &&
+					bytes.Compare(addrs.GetCoinType(), address.GetCoinType()) == 0 {
+					return fmt.Errorf("Address with value: %s  and Cointype: %s alredy exist",
+						address.GetValue(), address.GetCoinType())
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// nameExists search an name in the list of contacts into the AddressBook.
+// If find the address return error, else return nil.
+func (ab *DB) nameExists(contact core.Contact, contacts []core.Contact) error {
+	for _, c := range contacts {
+		if dataContact, ok := c.(*Contact); ok {
+			if bytes.Compare(contact.(*Contact).Name, dataContact.Name) == 0 {
+				return fmt.Errorf(" Contact with name: %s alredy exist", contact.(*Contact).Name)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (ab *DB) insertConfig(secType int, hash, entropy []byte) error {
+	tx, err := ab.db.Begin(true)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := tx.Rollback(); err != nil && err != bolt.ErrTxClosed {
+			logrus.Fatal(err)
+		}
+	}()
+
+	_, err = tx.CreateBucketIfNotExists(dbAddrsBookBkt)
+	if err != nil {
+		return err
+	}
+	_, err = tx.CreateBucketIfNotExists(dbConfigBkt)
+	if err != nil {
+		return err
+	}
+
+	bConf := tx.Bucket(dbConfigBkt)
+	if err := bConf.Put([]byte("hash"), hash); err != nil {
```

Comment:
If possible, create constants for this words , example:
```golang
const = (
    Hash           = "hash"
    Entropy       = "entropy"
    sectionType = "secType"
)
```
But this with names according to functions